app/controllers/resources_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class ResourcesController < ApplicationController
+  # before_action :authorize_request
+
+  # POST /objects
+  def create
+    register_params
+    response = Dor::Services::Client.objects.register(params: register_params)
+    workflow_client.create_workflow_by_name(response[:pid], 'accessionWF')
+
+    render json: { druid: response[:pid] }, status: :created
+  rescue Dor::Services::Client::ConnectionFailed => e
+    render json: build_error('Unable to reach dor-services-app', e), status: :gateway_timeout
+  end
+
+  private
+
+  def workflow_client
+    Dor::Workflow::Client.new(url: Settings.workflow.url,
+                              logger: Rails.logger,
+                              timeout: 60)
+  end
+
+  # @return [Hash] the parameters used to register an object.
+  def register_params
+    reg_params = {
+      object_type: 'object',
+      admin_policy: 'TODO: what policy?'
+    }
+    reg_params[:label] = params[:collection_title].presence || ':auto'
+    reg_params[:rights] = if reg_params[:label] == ':auto'
+                            params[:collection_rights_catkey]
+                          else
+                            params[:collection_rights]
+                          end
+    reg_params[:rights] &&= reg_params[:rights].downcase
+    col_catkey = params[:collection_catkey] || ''
+    reg_params[:metadata_source] = col_catkey.blank? ? 'label' : 'symphony'
+    reg_params[:other_id] = "symphony:#{col_catkey}" if col_catkey.present?
+    reg_params
+  end
+
+  # JSON-API error response
+  def build_error(msg, err)
+    {
+      json: {
+        errors: [
+          {
+            "status": '504',
+            "title": msg,
+            "detail": err.message
+          }
+        ]
+      },
+      content_type: 'application/vnd.api+json',
```

Comment:
Why not `application/json`?

Reply:
because this is  the registered mimetype for JSON api https://jsonapi.org/

Reply:
Cool. To help a dev not familiar with JSON-API, a link in the comment in L44 would be helpful to make it clear it is an external spec.